rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -784,12 +796,34 @@ impl CallContextManager {
     pub fn unregister_callback(&mut self, callback_id: CallbackId) -> Option<Arc<Callback>> {
         self.callbacks.remove(&callback_id).map(|callback| {
             self.stats.on_unregister_callback(&callback);
+            if callback.deadline != NO_DEADLINE {
+                self.unexpired_callbacks
+                    .remove(&(callback.deadline, callback_id));
+            }
             debug_assert!(self.stats_ok());
 
             callback
         })
     }
 
+    /// Returns the IDs of all best-effort callbacks whose deadlines have expired
+    /// since the previous call to this method, given the current time.
+    ///
+    /// Note: A given callback will be returned at most once by this function.
+    #[allow(dead_code)]
+    pub(super) fn expired_callbacks(
+        &mut self,
+        now: CoarseTime,
+    ) -> impl Iterator<Item = CallbackId> {
+        const MIN_CALLBACK_ID: CallbackId = CallbackId::new(0);
+
+        let mut expired_callbacks = self.unexpired_callbacks.split_off(&(now, MIN_CALLBACK_ID));
```

Comment:
It is a bit weird to call this `expired_callbacks` here because `split_off` will return a set of all callbacks with a deadline `>= now`, i.e., the ones that did not expire yet, and only the `std::mem::swap` will put things into the right place. 

Maybe consider renaming this to something else or add a comment to clarify this.

Reply:
Looked into the documentation of `std::mem::swap()` and it suggested `std::mem::replace()` as an alternative. It's still 3 steps and we now have two termporary variables, but it's a lot nicer.

Thanks.